metadata-service/restli-client/src/main/java/com/linkedin/entity/client/RestliEntityClient.java

```diff
@@ -681,4 +686,24 @@ public void rollbackIngestion(@Nonnull String runId, @Nonnull final Authenticati
     final RunsDoRollbackRequestBuilder requestBuilder = RUNS_REQUEST_BUILDERS.actionRollback().runIdParam(runId).dryRunParam(false);
     sendClientRequest(requestBuilder, authentication);
   }
+
+  // Creates new Filter from a map of Criteria by removing null-valued Criteria and using EQUAL condition (default).
+  @Nonnull
+  public static Filter newFilter(@Nullable Map<String, String> params) {
+    if (params == null) {
+      return new Filter().setOr(new ConjunctiveCriterionArray());
+    }
+    CriterionArray criteria = params.entrySet()
+        .stream()
+        .filter(e -> Objects.nonNull(e.getValue()))
+        .map(e -> newCriterion(e.getKey(), e.getValue(), Condition.EQUAL))
+        .collect(Collectors.toCollection(CriterionArray::new));
+    return new Filter().setOr(
+        new ConjunctiveCriterionArray(ImmutableList.of(new ConjunctiveCriterion().setAnd(criteria))));
+  }
+
+  @Nonnull
+  public static Criterion newCriterion(@Nonnull String field, @Nonnull String value, @Nonnull Condition condition) {
+    return new Criterion().setField(field).setValue(value).setCondition(condition);
+  }
```

Comment:
just curios - why duplicate these from QueryUtils to here?

Reply:
Query Utils are unfortunately inside of metadata-io which should be a library that is NOT exposed to folks who just want to use a RestliEntityClient... It's too heavy. So we've removed this dependency completely and as part of that we had to duplicate this shared logic. That being said, I think it'd be better moving forward to establish a module where truly "shared" libraries can reside. I'm going to add a TODO here